src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -309,9 +310,15 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                 services.AddOptions<ScaleOptions>()
                     .Configure<IConfiguration>((o, c) =>
                     {
-                        c.GetSection(ConfigurationSectionNames.JobHost)
-                         .GetSection(ConfigurationSectionNames.Scale)
-                         .Bind(o);
+                        var scaleSection = c.GetSection(ConfigurationSectionNames.JobHost).GetSection(ConfigurationSectionNames.Scale);
+                        if (scaleSection.Exists())
```

Comment:
Rather than this new conditional logic here, I think you should leave this code as is, and register an options setup (e.g. see ConcurrencyOptionsSetup). That's the way this is usually done.

Reply:
Created ScaleOptionsSetup.